src/d/d_a_leaflift_static.cpp

```diff
@@ -4,19 +4,43 @@
 //
 
 #include "d/actor/d_a_leaflift.h"
+#include "d/d_com_inf_game.h"
 #include "dolphin/types.h"
 
 const f32 daLlift_c::m_height = 560.0f;
 const f32 daLlift_c::m_max_speed = 10.0f;
 const f32 daLlift_c::m_min_speed = 5.0f;
 
 /* 800690E4-80069100       .text checkEndDownLift__9daLlift_cFv */
-void daLlift_c::checkEndDownLift() {
-    /* Nonmatching */
+s32 daLlift_c::checkEndDownLift() {
+    return (((current.pos.y <= home.pos.y) << 1) << 0x1c) >> 0x1d;
```

Comment:
This looks like a simple comparison that the compiler did a branchless optimization to. Have you tried using https://roeming.github.io/C-Math-Evaluator/ to help you simplify it? Keep in mind you may also need to mess with the return type of this function too, like BOOL or bool

Reply:
This still applies